src/operators/aggregate/aggregate.cc

```diff
@@ -353,40 +286,48 @@ void Aggregate::InsertGroupColumns(std::vector<int> group_id, int agg_index) {
   // Loop over columns in group builder, and append one of its unique values to
   // its builder.
   for (std::size_t i = 0; i < group_type_->num_fields(); ++i) {
-    switch (group_type_->field(i)->type()->id()) {
-      case arrow::Type::STRING: {
-        // Downcast the column's builder
-        auto col_group_builder =
-            (arrow::StringBuilder*)(group_builder_->child(i));
-        // Downcast the column's unique_values
-        auto col_unique_values =
-            std::static_pointer_cast<arrow::StringArray>(unique_values_[i]);
-
-        // Append one of the unique values to the column's builder.
-        status = col_group_builder->Append(
-            col_unique_values->GetString(group_id[i]));
-        evaluate_status(status, __FUNCTION__, __LINE__);
-        break;
-      }
-      case arrow::Type::INT64: {
-        // Downcast the column's builder
-        auto col_group_builder =
-            (arrow::Int64Builder*)(group_builder_->child(i));
-        // Downcast the column's unique_values
-        auto col_unique_values =
-            std::static_pointer_cast<arrow::Int64Array>(unique_values_[i]);
-
-        // Append one of the unique values to the column's builder.
-        status =
-            col_group_builder->Append(col_unique_values->Value(group_id[i]));
-        evaluate_status(status, __FUNCTION__, __LINE__);
-        break;
-      }
-      default: {
-        throw std::runtime_error("Cannot insert unsupported aggregate type: " +
-                                 group_type_->field(i)->type()->ToString());
+    // Handle the insertion of record.
+    auto insert_handler = [&, this]<typename T>(T * ptr_) {
+      // TODO: Add support for other types.
+      //  Right now we only support types that have builder and array
+      //  in the type trait. It should be obvious that we can:
+      //  - [1] Get builder from field spec
+      //  - [2] Get array for most cases (except for Dict and Extension).
+      constexpr bool is_supported_type = std::conjunction_v<
+          has_builder_type<T>, has_array_type<T>,
+          isNotOneOf<T, arrow::ExtensionType, arrow::DictionaryType>>;
+
+      if constexpr (is_supported_type) {
+        using BuilderType = typename arrow::TypeTraits<T>::BuilderType;
+        using ArrayType = typename arrow::TypeTraits<T>::ArrayType;
```

Comment:
I am thinking can we move all the aliases to the common file like type_*.h

Reply:
It will also be great if the type has a prefix "Arrow" (Optional) - Just for better readability

Reply:
Fixed in the new commit. Now we get 
```c++
using ArrayType = GetArrowArrayType<T>;
```